### ISSUE TITLE
Problem with app router when used with nextra & i18n

### ISSUE THREAD

**Issue by debsouryadatta** [shuding/nextra#2814]
I am using nextra docs in pages router and my main site on the app router, but when I tried to implement the i18n support for the nextra, I updated the nextconfig file, after that i am facing issues in the app directory routing, the pages in the app dir are not opening up(showing 404 error). What is the reason for this and how can I fix it?

**Comment by arno-fukuda** [comment]
https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher

You need to adjust the middleware to filter out routes you have in /app

**Comment by debsouryadatta** [comment]
> https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher
> 
> You need to adjust the middleware to filter out routes you have in /app

![Screenshot from 2024-03-30 21-46-38](https://github.com/shuding/nextra/assets/91617309/213c8d8b-bfd9-4bec-b47e-f1b4f5d6e959)

When i uncomment the commented lines from the next config, the app router breaks giving a 404 error, i am not able to use both pages router and the app router when i enable i18n init.
I was going through the middleware thing but couldn't find any solution, could you help me for the same

**Comment by arno-fukuda** [comment]
What is the request URL difference between when its working and returning 404?

**Comment by debsouryadatta** [comment]
> What is the request URL difference between when its working and returning 404?

Request URL is the same for both, even the "/" route gives a 404 error when I uncomment those lines for i18n

**Comment by arno-fukuda** [comment]
> > What is the request URL difference between when its working and returning 404?
> 
> Request URL is the same for both, even the "/" route gives a 404 error when I uncomment those lines for i18n

Can you post a CodeSandbox with your app?

**Comment by debsouryadatta** [comment]
> > > What is the request URL difference between when its working and returning 404?
> > 
> > 
> > Request URL is the same for both, even the "/" route gives a 404 error when I uncomment those lines for i18n
> 
> Can you post a CodeSandbox with your app?

Here you go, Sandbox link - https://codesandbox.io/p/devbox/sad-brattain-g2zf2h

**Comment by arno-fukuda** [comment]
> > > > What is the request URL difference between when its working and returning 404?
> > > 
> > > 
> > > Request URL is the same for both, even the "/" route gives a 404 error when I uncomment those lines for i18n
> > 
> > 
> > Can you post a CodeSandbox with your app?
> 
> Here you go, Sandbox link - https://codesandbox.io/p/devbox/sad-brattain-g2zf2h


middleware.js
```
import { locales as localeMiddleware } from "nextra/locales";
import { parse } from "url";

export const middleware = (req) => {
  const parsedUrl = parse(req.url, true);
  const path = parsedUrl.pathname;
  if (path !== "/") {
    console.log(req);
    return localeMiddleware(req);
  }
};

export const config = {
  matcher: ["/", "/((?!_next/static|_next/image).*)"],
};
```

next.config.mjs
```
import nextra from "nextra";

/** @type {import('next').NextConfig} */
const nextConfig = {
  i18n: {
    locales: ["en-US", "zh-CN", "de-DE"],
    defaultLocale: "en-US",
  },
};

const withNextra = nextra({
  theme: "nextra-theme-docs",
  themeConfig: "./theme.config.jsx",
});

export default withNextra(nextConfig);
```

You should be able to navigate to all pages now.

* /
* /docs/index
* /docs/tailwind/margin

tldr, don't run the middleware on routes used in /app

**Comment by debsouryadatta** [comment]
> > > > > What is the request URL difference between when its working and returning 404?
> > > > 
> > > > 
> > > > Request URL is the same for both, even the "/" route gives a 404 error when I uncomment those lines for i18n
> > > 
> > > 
> > > Can you post a CodeSandbox with your app?
> > 
> > 
> > Here you go, Sandbox link - https://codesandbox.io/p/devbox/sad-brattain-g2zf2h
> 
> middleware.js
> 
> ```
> import { locales as localeMiddleware } from "nextra/locales";
> import { parse } from "url";
> 
> export const middleware = (req) => {
>   const parsedUrl = parse(req.url, true);
>   const path = parsedUrl.pathname;
>   if (path !== "/") {
>     console.log(req);
>     return localeMiddleware(req);
>   }
> };
> 
> export const config = {
>   matcher: ["/", "/((?!_next/static|_next/image).*)"],
> };
> ```
> 
> next.config.mjs
> 
> ```
> import nextra from "nextra";
> 
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   i18n: {
>     locales: ["en-US", "zh-CN", "de-DE"],
>     defaultLocale: "en-US",
>   },
> };
> 
> const withNextra = nextra({
>   theme: "nextra-theme-docs",
>   themeConfig: "./theme.config.jsx",
> });
> 
> export default withNextra(nextConfig);
> ```
> 
> You should be able to navigate to all pages now.
> 
> * /
> * /docs/index
> * /docs/tailwind/margin
> 
> tldr, don't run the middleware on routes used in /app

Thanks a lot, now its working in the pages router but the main site in app router breaks,
How not to run the middleware on routes used in /app?
![Screenshot from 2024-03-31 12-36-09](https://github.com/shuding/nextra/assets/91617309/579d55e8-cf6a-4b76-9b5b-d944ec61fbc2)

**Comment by debsouryadatta** [comment]
@arno-fukuda Thanks a lot, now its working in the pages router but the main site in app router breaks,
How not to run the middleware on routes used in /app?
![Screenshot from 2024-03-31 12-36-09](https://github.com/shuding/nextra/assets/91617309/ef0abf54-7b74-4ea6-8521-482ecbd40e07)

**Comment by arno-fukuda** [comment]
```
// middleware.js

import { locales as localeMiddleware } from "nextra/locales";
import { parse } from "url";

export const middleware = (req) => {
  const parsedUrl = parse(req.url, true);
  const path = parsedUrl.pathname;

// here you need to conditionally filter out everything in your /app router, where we don't want the middleware to run
  if (path !== "/" && !path.startsWith("/main") {
    return localeMiddleware(req);
  }
};

// the matcher seems to only work on on the pages served for the pages router. You could also omit it if not needed.
export const config = {
  matcher: ["/", "/((?!_next/static|_next/image).*)"],
};

**Comment by debsouryadatta** [comment]
@arno-fukuda Thanks a lot, it's working fine now🙌, really a good maintainer✌️

**Comment by arno-fukuda** [comment]
@debsouryadatta  please close the issue.
